src/modules/adaptors/Frax/CollateralFTokenAdaptor.sol

```diff
@@ -0,0 +1,226 @@
+
+    constructor(address _frax) {
+        FRAX = ERC20(_frax);
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("FraxLend Collateral fToken Adaptor V 0.1"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice User deposits collateralToken to Fraxlend pair
+     * @param assets the amount of assets to lend on FraxLend
+     * @param adaptorData adaptor data containing the abi encoded fToken
+     * @dev configurationData is NOT used
+     * TODO: EIN write implementation code for base function
+     */
+    function deposit(
+        uint256 assets,
+        bytes memory adaptorData,
+        bytes memory
+    ) public override {
+        // use addCollateral() from fraxlendCore.sol
+    }
+
+    /**
+     * @notice User withdraws are NOT allowed from this position.
+     * TODO: EIN write implementation code for base function
+     */
+    function withdraw(
+        uint256,
+        address,
+        bytes memory,
+        bytes memory
+    ) public pure override {}
+
+    /**
+     * @notice This position is a debt position, and user withdraws are not allowed so
+     *         this position must return 0 for withdrawableFrom.
+     * TODO: EIN write implementation code for base function
+     */
+    function withdrawableFrom(bytes memory, bytes memory) public pure override returns (uint256) {
+        return 0; // TODO: see AaveATokenAdaptor for idea of what to do here. Not sure it applies.
+    }
+
+    /**
+     * @notice Returns the cellar's balance of the collateralAsset position.
+     * @param adaptorData the collateral asset deposited into Fraxlend
+     * NOTE: CRISPY QUESTION - TODO: confirm that this works... EIN - FraxlendCore doesn't have an interface to access the getter. So we'll have to have it here.
+     * TODO: confirm that there is no need to typeCast on the decoded adaptorData when needing address, ERC20 or vice versa.
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        ICollateralFToken fToken = ICollateralFToken(abi.decode(adaptorData, (address))); // TODO: CRISPY QUESTION - could change adaptorData to be collateralToken but then it won't be consistent with rest of fraxlend adaptors.
+        return fToken.userCollateralBalance(msg.sender) + fToken.collateralContract().balanceOf(msg.sender); // reports balance of collateral provided to protocol.
```

Comment:
I didn't realize this but, we'll need `collateral` as a `erc20Adaptor` trusted position asset.  It is accounted then with that adaptor for any lingering `collateral` ERC20 balance for the cellar.